hermes-common/src/main/java/pl/allegro/tech/hermes/domain/topic/schema/SchemaRepository.java

```diff
@@ -0,0 +1,114 @@
+package pl.allegro.tech.hermes.domain.topic.schema;
+
+import com.google.common.base.Ticker;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListenableFutureTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.SchemaSource;
+import pl.allegro.tech.hermes.api.Topic;
+
+import javax.inject.Inject;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class SchemaRepository<T> {
+
+    private static final Logger logger = LoggerFactory.getLogger(SchemaRepository.class);
+
+    private final LoadingCache<Topic, SchemaWithSource> schemaCache;
+    private final SchemaCompiler<T> schemaCompiler;
+
+    @Inject
+    public SchemaRepository(SchemaSourceProvider schemaRepository, ExecutorService reloadSchemaSourceExecutor, SchemaCompiler<T> schemaCompiler) {
+        this(schemaRepository, reloadSchemaSourceExecutor, Ticker.systemTicker(), schemaCompiler);
+    }
+
+    SchemaRepository(SchemaSourceProvider schemaSourceProvider, ExecutorService reloadSchemaSourceExecutor, Ticker ticker, SchemaCompiler<T> schemaCompiler) {
+        this.schemaCompiler = schemaCompiler;
+        this.schemaCache = CacheBuilder
+                .newBuilder()
+                .ticker(ticker)
+                .refreshAfterWrite(10, TimeUnit.MINUTES)
+                .expireAfterWrite(24, TimeUnit.HOURS)
+                .build(new CacheLoader<Topic, SchemaWithSource>() {
```

Comment:
i think it should be extracted to another place eg private factory method or external class